TheaterSchedule.BLL/Services/SettingsService.cs

```diff
@@ -34,37 +36,46 @@ public SettingsDTO LoadSettings(string phoneId)
             Settings settings = settingsRepository.GetSettingsByPhoneId(phoneId);
             if (settings != null)
             {
-                settingsRequest = new SettingsDTO() { LanguageCode = settings.Language.LanguageCode};
-            }         
+                settingsRequest = new SettingsDTO() { LanguageCode = settings.Language.LanguageCode };
+            }
             return settingsRequest;
         }
 
         public void StoreSettings(string phoneId, SettingsDTO settingsRequest)
         {
-            Language language = languageRepository.GetLanguageByName(settingsRequest.LanguageCode);
-            if (language == null)
+            try
             {
-                throw new ArgumentException("Invalid language");
-            }
+                Language language = languageRepository.GetLanguageByName(settingsRequest.LanguageCode);
+                if (language == null)
+                {
+                    throw new HttpStatusCodeException(HttpStatusCode.NotFound, "Invalid language");
+                }
 
-            Settings settings = settingsRepository.GetSettingsByPhoneId(phoneId);
-            if (settings != null)
-            {
-                settings.Language = language;
-            }
-            else
-            {
-                Settings newSettings = new Settings { Language = language };
-                settingsRepository.Add(newSettings);
 
-                accountRepository.Add(new Account
+                Settings settings = settingsRepository.GetSettingsByPhoneId(phoneId);
+                if (settings != null)
+                {
+                    settings.Language = language;
+                }
+                else
                 {
-                    PhoneIdentifier = phoneId,
-                    Settings = newSettings
-                });
+                    Settings newSettings = new Settings { Language = language };
+                    settingsRepository.Add(newSettings);
+
+                    accountRepository.Add(new Account
+                    {
+                        PhoneIdentifier = phoneId,
+                        Settings = newSettings
+                    });
+                }
+
+                theaterScheduleUnitOfWork.Save();
+            }
+            catch (Exception ex)
+            {
+                throw  ex;
```

Comment:
by calling `throw ex ` you will loose stack trace of exception. Use throw; to pass exception upper

Reply:
Main goal of exception middleware is handling of errors without writing in each place try catch statement. all exceptions will be handled in single place